staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -215,10 +215,31 @@ type GroupResources struct {
 	// The empty string represents the core API group.
 	// +optional
 	Group string
-	// Resources is a list of resources within the API group.
-	// Any empty list implies every resource kind in the API group.
+	// Resources is a list of resources within the API group. Specific subresources
+	// can be matched using a "/" to indicate the subresource. For example "pods" matches
+	// requests to "pods" and "pods/log" but "pods/log" only matches "pods/log".
```

Comment:
And pods/ matches only the main resource?

Reply:
pods/ doesn't match anything :)

Reply:
Maybe it should? I only want to audit spec updates, not millions of status updates. How?

Reply:
I see a valid use case for each of the following

- auditing all subresources
- auditing only the main resource
- auditing only the subset of subresources

Maybe we should create a separate API object to fully express all cases? E.g.

```go
type SubresourcePolicyRuleType int

const (
  Whitelist SubresourcePolicyRuleType = iota
  Blacklist
)

type SubresourcePolicyRule struct {
  Type SubresourcePolicyRuleType
  Subresources []string
}
```

or the same fields inlined in the `PolicyRule`. Checking this will be easier also

Reply:
...or just match `pods/` with the main resource, that's also reasonable

Reply:
That was my expectation: `pods` for all, `pods/` for the main resource, `pods/log` for log. I don't think we need big subresource resource for that.

Reply:
that seems odd to me... I'd expect `pods` to just be the main resource, and something involving a `/` (`pods/*`?) to be all subresources

Reply:
I think switching back to the RBAC implementation where "pods" only matches the top level resource is reasonable. In the discussion above we made "pods" match all subresources because that seemed like a reasonable default, but it actually doesn't seem like it is.